### PR TITLE
Update deletionTimestamp with information about finalizer effect

### PR DESCRIPTION
**Pull request by wackxu**
**What this PR does / why we need it**:

ref https://github.com/kubernetes/community/pull/1177

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:Fixes # https://github.com/kubernetes/website/issues/6128

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go [hunk 1]
```diff
@@ -177,11 +177,13 @@ type ObjectMeta struct {
 	// DeletionTimestamp is RFC 3339 date and time at which this resource will be deleted. This
 	// field is set by the server when a graceful deletion is requested by the user, and is not
 	// directly settable by a client. The resource is expected to be deleted (no longer visible
-	// from resource lists, and not reachable by name) after the time in this field. Once set,
-	// this value may not be unset or be set further into the future, although it may be shortened
-	// or the resource may be deleted prior to this time. For example, a user may request that
-	// a pod is deleted in 30 seconds. The Kubelet will react by sending a graceful termination
-	// signal to the containers in the pod. After that 30 seconds, the Kubelet will send a hard
+	// from resource lists, and not reachable by name) after the time in this field except when
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: prefer "...after the time in this field, once the `finalizers` list is empty. As long as the `finalizers` list contains items, deletion is blocked."

Comment (wackxu) [reply staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: @liggitt Done, PTAL

**Pull request merged**